firedrake/function.py

```diff
@@ -482,6 +482,14 @@ def __idiv__(self, expr):
 
     __itruediv__ = __idiv__
 
+    def __float__(self):
+
+        if all((self.ufl_element().family() == "Real",
+                self.function_space().shape == ())):
```

Comment:
Strange way of writing `self.ufl_element().family() == "Real" and self.function_space().shape == ()` ?

Reply:
This avoids PEP8 idiocy. The alternative:
```
if self.ufl_element().family() == "Real" \
    and self.function_space().shape == ():
```
Fails PEP8 because the indentation on the second line of the if statement is the same as the indentation in the block.

Reply:
Arguably 87 characters is not so long that you would need to break the line. If you go for 80 character wrapping, [`black`](https://pypi.org/project/black/) produces:
```python
    def __float__(self):
        if (
            self.ufl_element().family() == "Real"
            and self.function_space().shape == ()
        ):
            return ...
```

I guess that's not the standard style in Firedrake right now.

I think the continuation line version works OK like this:
```python
    def __float__(self):
        if self.ufl_element().family() == "Real" \
           and self.function_space().shape == ():
            return self
```

Reply:
The latter version fails flake8. I've done the black version.